exercises/protein-translation/protein_translation_test.go

```diff
@@ -1,98 +1,85 @@
 package protein
 
 import (
+	"errors"
 	"reflect"
 	"testing"
 )
 
 type codonCase struct {
-	input         string
-	expected      string
-	errorExpected error
+	input    string
+	expected string
 }
 
-var codonTestCases = []codonCase{
+var codonSuccessCases = []codonCase{
 	{
 		"AUG",
 		"Methionine",
-		nil,
 	},
 	{
 		"UUU",
 		"Phenylalanine",
-		nil,
 	},
 	{
 		"UUC",
 		"Phenylalanine",
-		nil,
 	},
 	{
 		"UUA",
 		"Leucine",
-		nil,
 	},
 	{
 		"UUG",
 		"Leucine",
-		nil,
 	},
 	{
 		"UCG",
 		"Serine",
-		nil,
 	},
 	{
 		"UAU",
 		"Tyrosine",
-		nil,
 	},
 	{
 		"UAC",
 		"Tyrosine",
-		nil,
 	},
 	{
 		"UGU",
 		"Cysteine",
-		nil,
 	},
 	{
 		"UGG",
 		"Tryptophan",
-		nil,
 	},
+}
+
+var codonInvalidBaseCases = []codonCase{
 	{
-		"UAA",
+		"ABC",
 		"",
-		ErrStop,
 	},
+}
+
+var codonStopCases = []codonCase{
 	{
-		"UAG",
+		"UAA",
 		"",
-		ErrStop,
 	},
 	{
-		"UGA",
+		"UAG",
 		"",
-		ErrStop,
 	},
 	{
-		"ABC",
+		"UGA",
 		"",
-		ErrInvalidBase,
 	},
 }
 
 func TestCodon(t *testing.T) {
-	for _, test := range codonTestCases {
+	for _, test := range codonSuccessCases {
 		actual, err := FromCodon(test.input)
-		if test.errorExpected != nil {
-			if test.errorExpected != err {
-				t.Fatalf("FAIL: Protein translation test: %s\nExpected error: %q\nActual error: %q",
-					test.input, test.errorExpected, err)
-			}
-		} else if err != nil {
+		if err != nil {
 			t.Fatalf("FAIL: Protein translation test: %s\nExpected: %s\nGot error: %q",
```

Comment:
It's easier to read test output if it's all on one line. We can save space here because we already know the package and the test in question; `go test` will output those for us before the failure message. All we need to see is the input and output. So we can write something like:

```suggestion
			t.Fatalf("FromCodon(%q): %v", err)
```